src/DPD-REACT/random_external_state.h

```diff
@@ -78,8 +78,8 @@
 namespace random_external_state {
 typedef uint64_t es_RNG_t;
 
-enum { MAX_URAND = 0xffffffffU };
-enum { MAX_URAND64 = 0xffffffffffffffffULL - 1 };
+constexpr uint32_t MAX_URAND = 0xffffffffU;
+constexpr uint64_t MAX_URAND64 = 0xffffffffffffffffULL - 1;
```

Comment:
@stanmoore1 this may impact the KOKKOS version of the DPD-REACT package, so I am a little uncomfortable making this change now. However, I want to make certain that somebody can compile LAMMPS when writing new code that requires the latest C++ standard and C++20 is much more strict with respect to types returned by enums (which are usually left to the discretion of the compiler). There are some similar, but not "deadly" issues within Kokkos source code (some .hpp files) as well. Will have to contact the Kokkos folks about that.

Reply:
@akohlmey I don't see those enums used anywhere in the KOKKOS version. KOKKOS uses its own pRNG so I think we are OK. I also have decent internal regression tests for DPD-REACT including for KOKKOS.

Reply:
It looks like that file was adapted from an old version of Kokkos, but it isn't used by the LAMMPS KOKKOS package. The file in the Kokkos repo has been updated, see https://github.com/kokkos/kokkos/blob/master/algorithms/src/Kokkos_Random.hpp#L725.

Reply:
@stanmoore1 thanks a lot for the feedback. After some consideration, both the changes to the enumerators feel too risky for the stable release. I'd rather have not all of the stable version of LAMMPS compatible with a C+20 or later mode compiler than some unexpected side effects for which we don't have enough time to see if they show up. So I will revert those now and reapply later. The other C++20 related changes are corrections for code that is not 100% correct even at C++11 level, so those will remain. It was only the more strict rules in C++20 that required them instead of tolerating the small deviation.